lib/ansible/modules/packaging/language/pip.py

```diff
@@ -519,13 +652,12 @@ def main():
                 extra_args = ' '.join(args_list)
 
         if extra_args:
-            cmd += ' %s' % extra_args
+            cmd.append(extra_args)
 
         if name:
-            for pkg in name:
-                cmd += ' %s' % _get_full_name(pkg, version)
+            cmd.extend(str(p) for p in packages)
         elif requirements:
-            cmd += ' -r %s' % requirements
+            cmd.extend(['-r', requirements])
         else:
```

Comment:
Don't use str().  Use to_native (in module code) instead.  (Use to_text in controller code).

I left another comment with how to use to_native

Reply:
where's str? did you mean the implicit type of `cmd` contents?